### ISSUE TITLE
ImportError: from cocotb import simulator

### ISSUE THREAD

**Issue by elgorwi** [cocotb/cocotb#1829]
Just pulled the latest from master and build is broken on windows mingw 32-bit.
All is working fine using version b08bf0ce74144886c17e6f824438890d08503a71
Could be related to this change:
https://github.com/cocotb/cocotb/commit/3a0bf9460515ae0c2269890b75ddbeaa80085cff

Here's a log when running the adder example: 
``` make SIM=modelsim
Traceback (most recent call last):
  File "C:\msys32\mingw32\bin\cocotb-config-script.py", line 11, in <module>
    load_entry_point('cocotb', 'console_scripts', 'cocotb-config')()
  File "C:/msys32/mingw32/lib/python3.8/site-packages/pkg_resources/__init__.py", line 490, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "C:/msys32/mingw32/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2859, in load_entry_point
    return ep.load()
  File "C:/msys32/mingw32/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2450, in load
    return self.resolve()
  File "C:/msys32/mingw32/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2456, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "C:/work/cocotb/cocotb/__init__.py", line 42, in <module>
    import cocotb.handle
  File "C:/work/cocotb/cocotb/handle.py", line 36, in <module>
    from cocotb import simulator
ImportError: DLL load failed while importing simulator: The specified module could not be found.
Makefile:47: /Makefile.sim: No such file or directory
make: *** No rule to make target '/Makefile.sim'.  Stop.
```
Any suggestions?

**Comment by eric-wieser** [comment]
Can you try uninstalling and reinstalling cocotb, in case you've ended up with a mixture of installations?

@ktbarrett, any ideas?

**Comment by themperek** [comment]
I can reproduce.

It looks this has to be removed (for 3.8 it does not do what we expect):
https://github.com/cocotb/cocotb/blob/bd4f5c4ef7827758c99a54ae8ad41f0f035c70a4/cocotb/_os_compat.py#L17-L19

**Comment by eric-wieser** [comment]
Why does this work in CI? Is our windows CI not 3.8?

**Comment by themperek** [comment]
I think conda is still 3.7 by default 

Maybe we can test more but would move to  azure/GitHub for this maybe?

**Comment by ktbarrett** [comment]
I guess I never checked, and I guess it worked by accident, the directory is wrong...

https://github.com/cocotb/cocotb/blob/bd4f5c4ef7827758c99a54ae8ad41f0f035c70a4/cocotb/_os_compat.py#L14

That should be `extra_dll_dir = os.path.dirname(__file__)`. The simulator module is not in 'libs'. Could you confirm @themperek?

**Comment by themperek** [comment]
The problem is not `simulator` module but dependencies from 'libs' folder.

**Comment by eric-wieser** [comment]
Is there any way to find out exactly which dll is failing to load?

**Comment by tpambor** [comment]
On Windows, probably something like rpath can be done using manifest files. See [https://bugs.python.org/issue35688#msg335667](https://bugs.python.org/issue35688#msg335667) for a solution for a similar issue.

**Comment by eric-wieser** [comment]
Perhaps the thing to do for the imminent 1.4 release is to just set `PATH` on all python versions, and come back to this later.

**Comment by tpambor** [comment]
![cocotb_trace](https://user-images.githubusercontent.com/1379478/82056764-b0aa6400-96c2-11ea-9508-67d52c512eed.png)
For the records here is a trace. It fails to load any dependencies in the libs folder

**Comment by eric-wieser** [comment]
Can you get the `AddDllDirectory` call in that trace too?

**Comment by tpambor** [comment]
Nope, these are only the syscalls. `AddDllDirectory` probably is handled in userspace.

**Comment by eric-wieser** [comment]
Which simulator are you using, @elgorwi?

**Comment by eric-wieser** [comment]
It's possible we need something like https://github.com/numpy/numpy/pull/13019

**Comment by elgorwi** [comment]
> 
> 
> I can reproduce.
> 
> It looks this has to be removed (for 3.8 it does not do what we expect):
> 
> https://github.com/cocotb/cocotb/blob/bd4f5c4ef7827758c99a54ae8ad41f0f035c70a4/cocotb/_os_compat.py#L17-L19

Thanks @themperek, commenting out these lines resolves the issue.

**Comment by elgorwi** [comment]
> 
> 
> Which simulator are you using, @elgorwi?

I'm using Modelsim SE 10.2c

**Comment by themperek** [comment]
@elgorwi 
>I can open a PR for the proposed fix, or do we want to proceed with #1836 ?

Not sure if we manage with #1836 for 1.4 release (next days) so probably fast "fix" would be good to have.

**Comment by elgorwi** [comment]
> 
> 
> @elgorwi
> 
> > I can open a PR for the proposed fix, or do we want to proceed with #1836 ?
> 
> Not sure if we manage with #1836 for 1.4 release (next days) so probably fast "fix" would be good to have.

OK, opened PR #1841

**Comment by eric-wieser** [comment]
Removing the 1.4 milestone since #1829 stops the bleeding
